modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementSuccessTest.java

```diff
@@ -120,62 +112,106 @@ public void testGetAllApplications() throws IOException {
+//
+//        // Extract application name from the response XML
+//        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+//        DocumentBuilder builder = factory.newDocumentBuilder();
+//        ByteArrayInputStream input = new ByteArrayInputStream(response.asString().getBytes(StandardCharsets.UTF_8));
+//        Document doc = builder.parse(input);
+//        doc.getDocumentElement().normalize();
+//        Element eElement = (Element) doc.getDocumentElement().getChildNodes();
+//        String appName = eElement.getElementsByTagName("ApplicationName").item(0).getTextContent();
+//
+//        if (StringUtils.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, tenant)) {
+//            Assert.assertEquals(appName, APPLICATION_IMPORT_APP_NAME_SUPER_TENANT,
+//                    "Application export response doesn't match.");
+//        } else {
+//            Assert.assertEquals(appName, APPLICATION_IMPORT_APP_NAME_TENANT,
+//                    "Application export response doesn't match.");
+//        }
+//    }
 
-        String importFilePath;
-        if (StringUtils.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, tenant)) {
-            importFilePath = this.getClass().getResource("sample-sp-import-super-tenant.xml").getPath();
-        } else {
-            importFilePath = this.getClass().getResource("sample-sp-import-tenant.xml").getPath();
-        }
-        String response = getResponseOfPostWithFile(APPLICATION_MANAGEMENT_API_BASE_PATH + APPLICATION_IMPORT_PATH,
-               importFilePath , "file");
+    @Test
+    public void createApplication() throws Exception {
 
-        ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
-        ApplicationModel applicationModel = jsonWriter.readValue(response, ApplicationModel.class);
-        // Assert created application name
-        Assert.assertNotNull(applicationModel);
-        if (StringUtils.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, tenant)) {
-            Assert.assertEquals(applicationModel.getName(), APPLICATION_IMPORT_APP_NAME_SUPER_TENANT);
-        } else {
-            Assert.assertEquals(applicationModel.getName(), APPLICATION_IMPORT_APP_NAME_TENANT);
-        }
+        String body = readResource("create-basic-application.json");
+        Response responseOfPost = getResponseOfPost(APPLICATION_MANAGEMENT_API_BASE_PATH, body);
+        responseOfPost.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
 
-        Assert.assertNotNull(applicationModel.getId());
-        // Extract app id using a regex
-        importedAppId = applicationModel.getId();
+        String location = responseOfPost.getHeader(HttpHeaders.LOCATION);
+        createdAppId = location.substring(location.lastIndexOf("/") + 1);
```

Comment:
shall we assertNotNull on 'location' before performing substring?

Reply:
Yes. We have assertedd that location header is null in line 184